### PR TITLE
Filter stack traces when rendered in expression info value renderer

### PR DESCRIPTION
**Pull request by Vampire**
This change is [Reviewable](https://reviewable.io/reviews/spockframework/spock/944)

### CODE REVIEW

**Comment by Vampire** [comment]
https://github.com/leonard84/spock/tree/implicit-assert-CodeArgument has many.
In Slack Leonard said that in his new tests the traces are not filtered, thus this PR to improve on this.
I thought a separate test might not be necessary?

**Comment by szpak** [comment]
In that case - having tests in a separate branch - I propose to heave @leonard84 merge your changes into his branch and later on to merge it all together.

**Comment by leonard84** [comment]
While my changes have some implicit coverage, it is good practice to add some unit test for only that feature.

**Comment by leonard84** [comment]
Well an unit-test would use `ExpressionInfoValueRenderer` with an exception directly. 

The exceptions in my branch were caused by the not yet correctly working changes, but you could easily simulate it if you throw an exception when calling `foo()` in `mock.bar({it.foo()})`.

**Comment by Vampire** [comment]
Ah, stupid me.
I tried to do an integration test instead of simply doing a unit test.
I added a unit test now that shows and verifies the new behavior.

File: spock-specs/src/test/groovy/org/spockframework/smoke/StackTraceFiltering.groovy [hunk 1]
```diff
@@ -309,6 +313,32 @@ apackage.ASpec|a feature|2
...
+      new StackTraceElement('org.codehaus.groovy.runtime.Foo', 'bar', null, 0),
+      new StackTraceElement('org.codehaus.groovy.reflection.Foo', 'bar', null, 0),
+      new StackTraceElement('org.codehaus.groovy.FooMetaClassBaz', 'bar', null, 0),
+      new StackTraceElement('groovy.FooMetaClassBaz', 'bar', null, 0),
+      new StackTraceElement('groovy.lang.MetaMethod', 'bar', null, 0),
+      new StackTraceElement('java.lang.reflect.Foo', 'bar', null, 0),
+      new StackTraceElement('sun.reflect.Foo', 'bar', null, 0),
+      new StackTraceElement('jdk.internal.reflect.Foo', 'bar', null, 0),
+      new StackTraceElement('org.spockframework.runtime.Foo', 'bar', null, 0),
+      new StackTraceElement('org.spockframework.runtime.foo.Baz', 'bar', null, 0)
```
Comment (szpak) [inline spock-specs/src/test/groovy/org/spockframework/smoke/StackTraceFiltering.groovy]: A nice way of stackTrace creation :).

It reminded me that I wanted to take a closer look at `StackWalker` in Java 9 :).

**Pull request merged**